### PR TITLE
Update exit signature

### PR DESCRIPTION
**Pull request by tsudmi**

### CODE REVIEW

File: sw_utils/ipfs.py [hunk 1]
```diff
@@ -12,20 +12,11 @@
...
-
-def to_json(data: bytes) -> list:
-    return [base64.b64encode(data).decode('ascii')]
-
-
-def from_json(data: list) -> bytes:
-    return base64.b64decode(data[0])
-
-
 class BaseUploadClient:
```
Comment (cyc60) [inline sw_utils/ipfs.py:15]:
```
from abc import ABC, abstractmethod

class BaseUploadClient(ABC):
    @abstractmethod
    async def upload_bytes(self, data: bytes) -> str:
        raise NotImplementedError
    @abstractmethod
    async def upload_json(self, data: dict | list) -> str:
        raise NotImplementedError
```

**Review by cyc60 — APPROVED** [review]

File: sw_utils/ipfs.py [hunk 2]
```diff
@@ -37,13 +28,16 @@ def __init__(
         self.username = username
         self.password = password
 
-    async def upload(self, data: bytes) -> str:
+    async def upload_bytes(self, data: bytes) -> str:
+        return await self.upload_json(_to_json(data))
```
Comment (cyc60) [inline sw_utils/ipfs.py:32]: We can use `add_bytes` method

File: sw_utils/ipfs.py [hunk 3]
```diff
@@ -152,3 +154,15 @@ async def fetch(self, ipfs_hash: str) -> bytes:
                 logger.error(e)
 
         raise RuntimeError(f'Failed to fetch IPFS data at {ipfs_hash}')
+
+
+def _strip_ipfs_prefix(ipfs_hash: str) -> str:
+    return ipfs_hash.replace('ipfs://', '').replace('/ipfs/', '')
+
+
+def _to_json(data: bytes) -> list:
```
Comment (cyc60) [inline sw_utils/ipfs.py:163]: confusing naming, what if data is a correct json dict?

**Review by cyc60 — CHANGES_REQUESTED** [review]

File: sw_utils/ipfs.py (hunk 3 above)
Comment (tsudmi) [reply sw_utils/ipfs.py:163]: That method must be used only for `bytes` type data. If JSON is uploaded, `upload_json` must be used.

File: sw_utils/ipfs.py (hunk 2 above)
Comment (tsudmi) [reply sw_utils/ipfs.py:32]: Don't quite get it. Do you suggest renaming upload_bytes to add_bytes?

File: sw_utils/ipfs.py (hunk 1 above)
Comment (tsudmi) [reply sw_utils/ipfs.py:15]: Fixed

**Review by cyc60 — DISMISSED** [review]

**Review by cyc60 — APPROVED** [review]

**Pull request merged**